packages/flame/lib/src/camera/camera_component.dart

```diff
@@ -45,6 +46,29 @@ class CameraComponent extends Component {
   })  : viewport = viewport ?? MaxViewport(),
         viewfinder = viewfinder ?? Viewfinder();
 
+  /// Create a camera that shows a portion of the game world of fixed size
+  /// [width] x [height].
+  ///
+  /// The viewport will be centered within the canvas, expanding as much as
+  /// possible on all sides while maintaining the [width]:[height] aspect ratio.
+  /// The zoom level will be set such in such a way that exactly [width] x
+  /// [height] pixels are visible within the viewport. The viewfinder will be
+  /// initially set up to show world coordinates (0, 0) at the center of the
+  /// viewport.
+  factory CameraComponent.withFixedResolution({
+    required World world,
+    required double width,
+    required double height,
+    List<Component>? hudComponents,
+  }) {
+    return CameraComponent(
+      world: world,
+      viewport: FixedAspectRatioViewport(aspectRatio: width / height)
+        ..addAll(hudComponents ?? []),
+      viewfinder: Viewfinder()..visibleGameSize = Vector2(width, height),
```

Comment:
I understand that `visibleGameSize` is the reason for this being a constructor instead of a class, but doesn't this field belong in the viewport class?

Reply:
Not really, because it controls the zoom level, which is part of the Viewfinder.

Think about it this way: Viewport is the window through which we look at the game World. The properties of a viewport are: its shape, size, and position on the screen. Once Viewport has a fixed position on the screen, everything that happens inside it is controlled by the Viewfinder: which game point to "look at", the angle of rotation, and the zoom level.

Reply:
But you agree that this feels inconsistent? Is there any other way we can design this so that the `FixedResolutionViewport` can be defined as a viewport?

Reply:
For a normal physical camera maybe, but for pretty much every other game engine that I know of you set the fixed resolution at the viewport.

Reply:
I think at least? I'm not well versed in other game engines.

Reply:
Maybe not, seems to be very different how this is done in different engines.